py/tests/beta/agent_data/test_agent_data_schema.py

```diff
@@ -268,6 +268,110 @@ def test_parse_extracted_field_metadata():
     assert result["email"].confidence == 0.92
 
 
+def test_parse_extracted_field_metadata_complex():
+    """Test parse_extracted_field_metadata with new citation format and reasoning field."""
+    raw_metadata = {
+        "title": {
+            "reasoning": "Combined key parametrics and construction from the datasheet for a structured title.",
+            "citation": [
+                {
+                    "page": 1,
+                    "matching_text": "PHE844/F844, Film, Metallized Polypropylene, Safety, 0.47 uF",
+                }
+            ],
+            "extraction_confidence": 0.9470628580889779,
+            "confidence": 0.9470628580889779,
+        },
+        "manufacturer": {
+            "reasoning": "VERBATIM EXTRACTION",
+            "citation": [{"page": 1, "matching_text": "YAGEO KEMET"}],
+            "extraction_confidence": 0.9997446550976602,
+            "confidence": 0.9997446550976602,
+        },
+        "features": [
+            {
+                "reasoning": "VERBATIM EXTRACTION",
+                "citation": [
+                    {"page": 1, "matching_text": "Features</td><td>EMI Safety"}
+                ],
+                "extraction_confidence": 0.9999308195540074,
+                "confidence": 0.9999308195540074,
+            },
+            {
+                "reasoning": "VERBATIM EXTRACTION",
+                "citation": [
+                    {"page": 1, "matching_text": "THB Performance</td><td>Yes"}
+                ],
+                "extraction_confidence": 0.8642493886452225,
+                "confidence": 0.8642493886452225,
+            },
+        ],
+        "dimensions": {
+            "length": {
+                "citation": [{"page": 1, "matching_text": "L</td><td>41mm MAX"}],
+                "extraction_confidence": 0.8986941382802304,
+                "confidence": 0.8986941382802304,
+            },
+            "width": {
+                "citation": [{"page": 1, "matching_text": "T</td><td>13mm MAX"}],
+                "extraction_confidence": 0.9999377974447091,
+                "confidence": 0.9999377974447091,
+            },
+            "reasoning": "VERBATIM EXTRACTION",
```

Comment:
@neerajprad having the reasoning on the parent object of a nested object like this is a little unintuitive, and makes parsing difficult. My parser effectively pushes it down the leaves. Could this perhaps be modified to push down the leaves by default?